template/src/App.tsx

```diff
@@ -1,25 +1,31 @@
+import { Flex, VStack } from '@chakra-ui/layout';
+import { Spinner } from '@chakra-ui/spinner';
 import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { useAuthContext } from './context/AuthContext';
+
+const UnAuthApp = React.lazy(() => import('./pages/index/UnAuthApp'));
+
+const AuthApp = React.lazy(() => import('./pages/index/AuthApp'));
```

Comment:
I prefer to use AuthenticatedApp and UnauthenticatedApp. AuthApp looks like it is the 'app' for user authentication instead of the one used when user is already authenticated.

Reply:
Indeed